### PR TITLE
Clean Back-office templates, part 3 - improve

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | This PR should be considered as a BC because of lot of HTML structure changes and they might be used by a module or anything else, but it's acceptable. The optimizations are only within the files itself, so it does not matter if anyone uses the old version in any form in their modules.
| Deprecations?     | no
| How to test?      | Go through Improve section in BO and see if nothing is broken by this PR. There should be no visible change.
| Possible impacts? | No bad impacts. :-)

### Description
- I went through all templates and removed 99% of unneeded boostrap code like `<div class="row"><div class="col">XXX</div></div>`. It will reduce render depth and complexity by several layers - easier to navigate, debug CSS, maybe faster to render on slower PCs.
- I fixed all places where somebody used single quotations or no quotations at all.
- I fixed all places where somebody combined row and column in same element, this is not allowed.
- I fixed all places where somebody used syntax line col-sm-12 col-md-12 or col-md-12 col-lg-6. The first one is not needed.
- Small fixes here and there - removed some container classes, fixed module position page responsivity a bit, fixed virtual product block responsivity etc...
It should produce no BC breaks, unless some 3rd party module was targetting some classes they shouldn't have OR they targeted it by child or parent.

### Visible fixes
I fixed the responsivity of module positions page a bit. The modules itself still need to be improved, but at least the basic layout is correct now. The card on the right switches position to the top on mobile.

module_desktop
module_mobile

### Templates concerned
Improve/Design
Improve/Design/Cms
Improve/Design/MailTheme
Improve/Design/Theme
Improve/International/Currency
Improve/International/Geolocation
Improve/International/Language
Improve/International/Localization
Improve/International/Tax
Improve/International/TaxRulesGroup
Improve/International/Translations
Improve/International/Zone
Improve/Payment/PaymentMethods
Improve/Payment/Preferences
Improve/Shipping/Carriers
Improve/Shipping/Preferences
Module
Module/Includes

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26062)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]
@Julievrz no wording changes

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/Theme/index.html.twig [hunk 1]
```diff
@@ -41,126 +41,124 @@
...
 
-            {% include '@PrestaShop/Admin/Improve/Design/Theme/Blocks/layouts_configuration.html.twig' %}
-          </div>
-        </div>
+        {% endembed %}
+
+        {% include '@PrestaShop/Admin/Improve/Design/Theme/Blocks/layouts_configuration.html.twig' %}
       </div>
     </div>
+
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/Theme/index.html.twig:161]: Extra line

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 2]
```diff
@@ -26,34 +26,58 @@
...
-      <div class="card-body">
-        <div class="card">
-          <form class="mt-2" id="position-filters">
-            <div class="container">
-              <div class="row">
-                <div class="row col-md-12 col-lg-6">
-                  <label class="form-control-label col-4 text-left mt-1">{{ 'Show' | trans({}, 'Admin.Actions') }}</label>
-                  <div class="col-8">
+    <div class="col-md-3">
+      <div class="card" id="modules-position-selection-panel">
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:37]: This block has been moved, is it on purpose? This is part of your modification to make the page more responsive?

File: src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig [hunk 3]
```diff
@@ -31,7 +31,8 @@
 
   <div class="card">
     <h3 class="card-header">
-      <i class="material-icons">file_copy</i> {{ 'Copy'|trans({}, 'Admin.Actions') }}
+      <i class="material-icons">file_copy</i>
+      {{ 'Copy'|trans({}, 'Admin.Actions') }}
     </h3>
 
     <div class="card-block row">
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:38]: Seems like this one has a useless `row` class no?

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Shipping/Preferences/Blocks/shipping_preferences_handling.html.twig [hunk 4]
```diff
@@ -27,33 +27,28 @@
...
+      <span class="help-box" 
+            data-container="body" 
+            data-toggle="popover" 
+            data-trigger="hover" 
+            data-placement="right" 
+            data-content="{{ 'If you set these parameters to 0, they will be disabled.'|trans({}, 'Admin.Shipping.Help') }} {{ 'Coupons are not taken into account when calculating free shipping.'|trans({}, 'Admin.Shipping.Help') }}" 
+            title="">
+      </span>
+    </h3>
+    <div class="card-block row">
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Shipping/Preferences/Blocks/shipping_preferences_handling.html.twig:44]: No `column` inside this `row`, is the class really useful here?

**Review by jolelievre — COMMENTED** [review]
Thanks @Hlavtox 
Just a few questions on this one, but most of it seems ok!

File: src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig (hunk 3 above)
Comment (Hlavtox) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:38]: @jolelievre That's a syntax of form blocks, @NeOMakinG will redo this part. It messes up the look of them if you remove the **row** class element.

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Shipping/Preferences/Blocks/shipping_preferences_handling.html.twig (hunk 4 above)
Comment (Hlavtox) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Shipping/Preferences/Blocks/shipping_preferences_handling.html.twig:44]: @jolelievre The same as other comment, will be tackled in Valentin's PR.

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig (hunk 2 above)
Comment (Hlavtox) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:37]: @jolelievre Exactly, added screenshots to PR description. 👍

File: src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig (hunk 3 above)
Comment (NeOMakinG) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:38]: Uh, did I said that? 😆  I don't remember

Comment (Hlavtox) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:38]: @NeOMakinG Don't pretend you don't know!! 😄 https://github.com/PrestaShop/PrestaShop/pull/25879

Comment (jolelievre) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/International/Translations/Blocks/copy_language.html.twig:38]: Ok, so I guess we will need both PRs to be merged to have everything clean
Good luck to the first one who gets his PR merge, but most of all for the one who will have to deal with conflicts 😅

**Review by jolelievre — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]
But it needs a rebase

**Comment by khouloudbelguith** [comment]
Hi @Hlavtox,

Can you please check the Conflicting files?

Thanks!

**Comment by Hlavtox** [comment]
@khouloudbelguith Rebased, QA please? :-)

**Review by NeOMakinG — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_grid.html.twig [hunk 5]
```diff
@@ -25,7 +25,7 @@
 {% set isModuleActive = module.database.active %}
 
 <div
-  class="module-item module-item-grid col-md-12 col-lg-6 col-xl-3 {% if origin == 'manage' and isModuleActive == '0' %}module-item-grid-isNotActive{% endif %}"
+  class="module-item module-item-grid col-lg-6 col-xl-3 {% if origin == 'manage' and isModuleActive == '0' %}module-item-grid-isNotActive{% endif %}"
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_grid.html.twig:28]:
```suggestion
  class="module-item module-item-grid col-lg-6 col-xl-3{% if origin == 'manage' and isModuleActive == '0' %} module-item-grid-isNotActive{% endif %}"
```

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig [hunk 6]
```diff
@@ -38,7 +38,8 @@
   ats.features,
   ats.badges is defined and ats.badges|length > 0 ? ats.badges : false
 %}
-<div class="modal-dialog module-modal-dialog">
+<div
+  class="modal-dialog module-modal-dialog">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig:42]:
```suggestion
<div class="modal-dialog module-modal-dialog">
```

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig [hunk 7]
```diff
@@ -69,12 +70,13 @@
...
-      <div class="col-md-12 module-big-cover">
+    <div class="modal-body module-modal-body">
+      <div class="module-big-cover">
         <img src="{% if cover.big is defined %}{{ cover.big }}{% else %}{{ notFoundImg }}{% endif %}"/>
       </div>
-      <div class="col-md-12 module-menu-readmore">
-        <nav class="navbar navbar-light">
+      <div class="module-menu-readmore">
+        <nav
+          class="navbar navbar-light">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig:79]:
```suggestion
        <nav class="navbar navbar-light">
```

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig [hunk 8]
```diff
@@ -108,7 +110,8 @@
             {# end tab list #}
           </ul>
         </nav>
-        <div class="tab-content">
+        <div
+          class="tab-content">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more_content.html.twig:114]:
```suggestion
        <div class="tab-content">
```

**Review by Progi1984 — COMMENTED** [review]
@Hlavtox 4 small comments and it's approved

**Comment by Hlavtox** [comment]
@Progi1984 Done! ;-)

**Review by NeOMakinG — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@hibatallahAouadni rebased

**Review by jolelievre — APPROVED** [review]
Thanks @Hlavtox

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

All LGTM, except when I launch the auto tests (UI/campaigns/functional/BO), I encounter these issues:
- Payment>Preferences>currencyRestrictions :x: 
- International>Localization>Localization>importLocalizationPack :x: 
- International>Localization>Localization>defaultCurrency :x: 
- International>Localization>Currencies>CreateUnofficialCurrency :x: 
- International>Taxes>Taxes>filterAndQuickEditTaxes :x: 
- International>Taxes>Taxes>taxOptionsForm :x: 
- International>Taxes>Taxes>sortAndPagination :x:
- International>Taxes>TaxRules>filterSortAndPagination :x: 

image

---
### Mobile 
Well done with **Positions** page :clap: 
image

How it was on develop and 1.7.8.0:
image

Ping @PrestaShop/prestashop-core-developers & @boubkerbribri wdyt about the issues encountred (only on PR)?

Thanks!

**Comment by boubkerbribri** [comment]
Thanks @hibatallahAouadni 
@nesrineabdmouleh or I will fix UI tests after @PrestaShop/qa-functional validate the PR (manually).

**Comment by Hlavtox** [comment]
Thanks everyone! I think it will be the same case for part 4 and part 5.

**Comment by nesrineabdmouleh** [comment]
> Thanks @hibatallahAouadni @nesrineabdmouleh or I will fix UI tests after @PrestaShop/qa-functional validate the PR (manually).

Thanks @hibatallahAouadni 
UI tests are green

**Review by NeOMakinG — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

For the encountered issues with UI test, I test it manually:

- Payment>Preferences>currencyRestrictions :heavy_check_mark: 
- International>Localization>Localization>importLocalizationPack :heavy_check_mark: 
- International>Localization>Localization>defaultCurrency :heavy_check_mark: 
- International>Localization>Currencies>CreateUnofficialCurrency :heavy_check_mark: 
- International>Taxes>Taxes>filterAndQuickEditTaxes :heavy_check_mark: 
- International>Taxes>Taxes>taxOptionsForm :heavy_check_mark: 
- International>Taxes>Taxes>sortAndPagination :heavy_check_mark: 
- International>Taxes>TaxRules>filterSortAndPagination :heavy_check_mark: 

So LGTM, QA :heavy_check_mark: 
@PrestaShop/qa-automation for the issues I encountered on develop it was due to the timeout, when I relaunched it, all tests are green :wink: 
But the issues encountred on PR is due to changing selectors, so I think when we merge this PR, the Nightly will be crashed :sweat: 

Thanks!

**Comment by Hlavtox** [comment]
@boubkerbribri @hibatallahAouadni I think it's time to refactor the tests, if they depend on `div > div > div` or use boostrap column classes. ;-)

**Comment by boubkerbribri** [comment]
> @boubkerbribri @hibatallahAouadni I think it's time to refactor the tests, if they depend on `div > div > div` or use boostrap column classes. ;-)

Hi @Hlavtox,
It's already checked and it's all good.
Thanks for the PR, it's time to merge :muscle:.

**Review by boubkerbribri — APPROVED** [review]
We did a double-check for UI tests, it's all good.
Thanks @Hlavtox and @hibatallahAouadni

**Comment by Hlavtox** [comment]
Nice!!!

Thanks @boubkerbribri @hibatallahAouadni 🚀

**Comment by jolelievre** [comment]
Seems like everything is good for this one!!

Thanks @Hlavtox @boubkerbribri @hibatallahAouadni and everyone involved!!

**Pull request merged**